### PR TITLE
06 - added pagination sorting and swagger ability

### PR DESCRIPTION
**Pull request by yevhenkukharchuk**

### CODE REVIEW

File: src/main/java/yevhen/bookstore/controller/BookController.java [hunk 1]
```diff
@@ -18,41 +21,49 @@
...
-    public List<BookDto> getAll() {
-        return bookService.findAll();
+    @Operation(summary = "Get all books", description = "Get a list of all available books")
+    public List<BookDto> getAll(Pageable pageable) {
+        return bookService.findAll(pageable);
     }
 
     @GetMapping("/{id}")
+    @Operation(summary = "Get one book by id", description = "Get one book by id")
     public BookDto getBookById(@PathVariable Long id) {
```
Comment (dmytrocherepov) [inline src/main/java/yevhen/bookstore/controller/BookController.java:39]: it is better to valid your id , like use @Positive

File: src/main/java/yevhen/bookstore/controller/BookController.java [hunk 2]
```diff
@@ -18,41 +21,49 @@
...
     @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Create a new book", description = "Create a new book")
     public BookDto createBook(@RequestBody @Valid CreateBookRequestDto requestDto) {
         return bookService.save(requestDto);
     }
 
     @PutMapping("/{id}")
+    @Operation(summary = "Update a book by id", description = "Update a book with certain id")
     public BookDto updateBookById(@RequestBody CreateBookRequestDto requestDto,
                                   @PathVariable Long id) {
```
Comment (dmytrocherepov) [inline src/main/java/yevhen/bookstore/controller/BookController.java:53]: same

File: src/main/java/yevhen/bookstore/controller/BookController.java [hunk 3]
```diff
@@ -18,41 +21,49 @@
...
         return bookService.updateBookById(requestDto, id);
     }
 
     @ResponseStatus(HttpStatus.NO_CONTENT)
     @DeleteMapping("/{id}")
+    @Operation(summary = "Delete one book by id", description = "Delete one book by id")
     public void deleteBookById(@PathVariable Long id) {
```
Comment (dmytrocherepov) [inline src/main/java/yevhen/bookstore/controller/BookController.java:60]: same

**Review by dmytrocherepov — APPROVED** [review]

**Review by dmytrocherepov — APPROVED** [review]

File: src/main/java/yevhen/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -18,41 +21,49 @@
...
         bookService.deleteBookById(id);
     }
 
     @GetMapping("/search")
+    @Operation(summary = "Filter books by parameters",
+            description = "Filter books by titles and authors")
     public List<BookDto> searchBooks(BookSearchParametersDto searchParameters) {
```
Comment (liliia-ponomarenko) [inline src/main/java/yevhen/bookstore/controller/BookController.java:67]: Should you add pageble here?

**Review by liliia-ponomarenko — APPROVED** [review]
Good job! But check my comment before merging ;)

**Pull request merged**